File: Notification System/src/MailNotifier.cpp
#include "MailNotifier.hpp"
#include <iostream>

void MailNotifier::sendNotification(const std::string& sensorId,
                                    const std::string& message,
                                    const std::string& timestamp) {
    std::cout << "[MAIL] " << timestamp
              << " | Capteur: " << sensorId
              << " | Message: " << message << std::endl;
    // TODO: implémenter envoi réel via SMTP
}

File: Notification System/notifiers/mail_notifier.py
from .notifier import Notifier


class MailNotifier(Notifier):

    def send_notification(self, sensor_id: str, message: str, timestamp: str):
        print(f"[MAIL] {timestamp} | Capteur: {sensor_id} | Message: {message}")
        # TODO: implémenter envoi réel via SMTP
